src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -199,6 +200,17 @@ await Parallel.ForEachAsync(resources, parallelOptions, async (resource, innerCt
                     var result = new DataStoreOperationOutcome(upsertOutcome);
                     results.AddOrUpdate(identifier, _ => result, (_, _) => result);
                 }
+                catch (CosmosException cme) when (cme.StatusCode == HttpStatusCode.Forbidden && cme.SubStatusCode == 1014)
+                {
+                    // https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/troubleshoot-forbidden#partition-key-exceeding-storage
+
+                    var exceptionWithNoTechnicalDetails = new PreconditionFailedException("Partition reached maximum size.");
+                    var exceptionForLogging = new PreconditionFailedException(exceptionWithNoTechnicalDetails.Message, cme);
```

Comment:
Why are two exceptions needed here? We don't strip out the technical details in other exceptions, they aren't included in the response message.

Reply:
Thanks for confirming @LTA-Thinking . I'm gonna refactor that part. My concern in this scenario was about sharing too much information.